detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/GoModCommandRunner.java

```diff
@@ -0,0 +1,87 @@
+package com.synopsys.integration.detectable.detectables.go.gomod;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.synopsys.integration.detectable.ExecutableTarget;
+import com.synopsys.integration.detectable.ExecutableUtils;
+import com.synopsys.integration.detectable.detectable.executable.DetectableExecutableRunner;
+import com.synopsys.integration.detectable.detectable.executable.ExecutableFailedException;
+
+public class GoModCommandRunner {
+    // java:S5852: Warning about potential DoS risk.
+    @SuppressWarnings({ "java:S5852" })
+    private static final Pattern GENERATE_GO_LIST_JSON_OUTPUT_PATTERN = Pattern.compile("\\d+\\.[\\d.]+"); // Example: "go version go1.17.5 darwin/amd64" -> ""
+    private static final String JSON_OUTPUT_FLAG = "-json";
```

Comment:
This rule of static strings seems inconsistently applied. Why are only some of them static strings when others are not? You use "-m" as a static flag belor. You use "list" and "mod" multiple times.

Reply:
I honestly don't like the static strings, but this is to appease sonar. I only added it were sonar complained. 
I'd be fine with the policy that if sonar suggests moving any strings, we move all the strings.